logicalplan/distribute.go

```diff
@@ -21,7 +21,7 @@ import (
 )
 
 var (
-	RewrittenExternalLabelWarning = errors.Newf("%s: rewriting an external label with label_replace could lead to unpredictable results", annotations.PromQLWarning.Error())
+	RewrittenExternalLabelWarning = errors.Newf("%s: rewriting an external label with label_replace optimizing the query for distributed execution", annotations.PromQLWarning.Error())
```

Comment:
The warning is now hard to read. Maybe: 

```suggestion
	RewrittenExternalLabelWarning = errors.Newf("%s: rewriting an external label with label_replace disable optimizations for query distributed execution", annotations.PromQLWarning.Error())
```